tests/wasi-fyi/fs_open_root.rs

```diff
@@ -0,0 +1,44 @@
+use std::ffi::CString;
+use std::fs::File;
+
+#[link(wasm_import_module = "wasi_snapshot_preview1")]
+extern "C" {
+    pub fn path_open(
+        fd: i32,
+        dirflags: i32,
+        path: i32,
+        path_len: i32,
+        oflags: i32,
+        fs_rights_base: i64,
+        fs_rights_inheriting: i64,
+        fdflags: i32,
+        result_fd: i32,
+    ) -> i32;
+}
+
+const ERRNO_INVAL: i32 = 28;
+
+fn main() {
+    unsafe {
+        let fd = 5;
```

Comment:
Does this `fd` change if the runtime pre-opens things differently? For example, it can be `6`, or `7`, or ...

Reply:
Yes, potentially. AFAIK wasmer is the only runtime that supplies additional preopens. Other runtimes mostly begins at 3 (after stdin, stdout, and stderr). With the [proposed update to wasi-afk](https://github.com/wasmerio/wasmer/issues/4771), we can add a function that calls `fd_prestat_get` and dynamically determines the base fd and share this helper function. I'll leave it up to you if you want to create a issue for that.